src/cdk/a11y/focus-trap/event-listener-inert-strategy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusTrapInertStrategy} from './focus-trap-inert-strategy';
+import {ConfigurableFocusTrap} from './configurable-focus-trap';
+
+/**
+ * Lightweight FocusTrapInertStrategy that adds a document focus event
+ * listener to redirect focus back inside the FocusTrap.
+ */
+export class EventListenerFocusTrapInertStrategy implements FocusTrapInertStrategy {
+  /** Focus event handler. */
+  private _listener: any;
+
+  /** Adds a document event listener that keeps focus inside the FocusTrap. */
+  preventFocus(focusTrap: ConfigurableFocusTrap): void {
+    this._listener = this._trapFocus.bind(null, focusTrap);
+    focusTrap.ngZone.runOutsideAngular(() => {
+      focusTrap.document.addEventListener('focus', this._listener, true);
```

Comment:
Should we have some logic that ensures that there's only one listener per document?

Reply:
That seems reasonable. It shouldn't happen if FocusTrap is working correctly, but I added a check to remove the current listener, if any, before adding another!